scanner/internal/httputil/transport_mux.go

```diff
@@ -0,0 +1,126 @@
+// Package httputil defines utility HTTP functions built specifically for Scanner's use cases.
+//
+// It is possible other StackRox components may benefit from these functions,
+// but that will be considered at a future time.
+package httputil
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/mtls"
+)
+
+var (
+	// defaultDialer is essentially copied from http.DefaultTransport from go1.20.10.
+	defaultDialer = net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+)
+
+type options struct {
+	denySensor bool
+
+	// These are here for testing purposes.
+	centralTransport http.RoundTripper
+	sensorTransport  http.RoundTripper
+}
+
+// TransportOption configures options for HTTP transport.
+type TransportOption func(o *options)
+
+// WithDenySensor configures whether the transport should deny traffic to Sensor.
+func WithDenySensor(deny bool) TransportOption {
+	return func(o *options) {
+		o.denySensor = deny
+	}
+}
+
+// TransportMux returns a http.RoundTripper which multiplexes the given default http.RoundTripper
+// with ones which support mTLS with StackRox services.
+//
+// It is assumed the StackRox services are in the same Kubernetes namespace.
+//
+// Only Central and Sensor are supported StackRox services at this time.
+func TransportMux(defaultTransport http.RoundTripper, opts ...TransportOption) (http.RoundTripper, error) {
+	var o options
+	for _, opt := range opts {
+		opt(&o)
+	}
+
+	return transportMux(defaultTransport, o)
+}
+
+func transportMux(defaultTransport http.RoundTripper, o options) (http.RoundTripper, error) {
+	centralTransport := o.centralTransport
+	if centralTransport == nil {
+		var err error
+		centralTransport, err = roxTransport(mtls.CentralSubject)
+		if err != nil {
+			return nil, fmt.Errorf("creating Central TLS config: %w", err)
+		}
+	}
+
+	sensorTransport := DenyTransport
+	if !o.denySensor {
+		sensorTransport = o.sensorTransport
+		if sensorTransport == nil {
+			var err error
+			sensorTransport, err = roxTransport(mtls.SensorSubject)
+			if err != nil {
+				return nil, fmt.Errorf("creating Sensor TLS config: %w", err)
+			}
+		}
+	}
+
+	// This is here instead of as a global variable for testing purposes.
+	namespace := env.Namespace.Setting()
+	centralHost := fmt.Sprintf("central.%s.svc", namespace)
+	sensorHost := fmt.Sprintf("sensor.%s.svc", namespace)
```

Comment:
Seems most of the indexer/matcher URI's are config file driven, should these be part of the config?

Reply:
Thought about it, but I decided against it. These fields aren't really configurable. They are pretty locked in. I think if we do want to make this configurable, though, we can in the future

Reply:
something like the repo2cpe URL, for example, is configurable because it made it easy to keep making progress while we were unable to reach central when testing. similarly, it allows us to use Scanner v4 without deploying the rest of Rox-services